src/test/java/io/confluent/connect/jdbc/source/JdbcSourceTaskUpdateTest.java

```diff
@@ -342,10 +423,38 @@ public void testAutoincrementRestoreOffset() throws Exception {
   }
 
   @Test
-  public void testTimestampRestoreOffset() throws Exception {
+  public void testTimestampRestoreNoVersionOffset() throws Exception {
+    TimestampIncrementingOffset offset = new TimestampIncrementingOffset(new Timestamp(10L), null);
+    testTimestampRestoreOffset(Collections.singletonMap(SINGLE_TABLE_PARTITION, offset.toMap()));
+  }
+
+  @Test
+  public void testTimestampRestoreVersionOneOffset() throws Exception {
+    TimestampIncrementingOffset offset = new TimestampIncrementingOffset(new Timestamp(10L), null);
+    testTimestampRestoreOffset(
+        Collections.singletonMap(SINGLE_TABLE_PARTITION_WITH_VERSION, offset.toMap())
+    );
+  }
+
+  @Test
+  public void testTimestampRestore() throws Exception {
+    TimestampIncrementingOffset oldOffset = new TimestampIncrementingOffset(
+        new Timestamp(8L),
+        null
+    );
     TimestampIncrementingOffset offset = new TimestampIncrementingOffset(new Timestamp(10L), null);
-    expectInitialize(Arrays.asList(SINGLE_TABLE_PARTITION),
-                     Collections.singletonMap(SINGLE_TABLE_PARTITION, offset.toMap()));
+    Map<Map<String, String>, Map<String, Object>> offsets = new HashMap<>();
+    offsets.put(SINGLE_TABLE_PARTITION_WITH_VERSION, offset.toMap());
+    offsets.put(SINGLE_TABLE_PARTITION, oldOffset.toMap());
+    testTimestampRestoreOffset(offsets);
```

Comment:
How about a comment here to say that we want to always use the offset with the latest protocol found?